advzip.1
.\" Process this file with
.\" groff -man -Tascii advzip.1
.\"
.TH AdvanceZIP 1 "01 Maggio 2002"
.SH NAME
advzip \- zip utility
.SH SYNOPSIS
.B advzip [-a, --add] [-x, --extract] [-l, --list]
.B [-z, --recompress] [-t, --test]
.B [-0, --shrink-0] [-1, --shrink-1] [-2, --shrink-2] [-3, --shrink-3]
.B [-N, --not-zip] [-p, --pedantic]
.B [-q, --quiet] [-h, --help] [-V, --version]
ARCHIVES... [FILES...]
.SH DESCRIPTION
The main purpose of this utility is to recompress and test the zip archives
to get the smallest possible size.
.PP
For the recompression the 7-Zip (www.7-zip.com) Deflate implementation is
used. This implementation generally gives 5-10% more compression than the
zLib Deflate implementation.
.PP
For experimental purpose also the 7-Zip LZMA algorithm is available with
the -N option. In this case the generated zips WILL NOT BE USEABLE by any other
program. To make them useable you need to recompress them without the -N option.
Generally this algorithm gives 10-20% more compression than the 7-Zip Deflate
implementation.
.SH OPTIONS
.TP
.B -a, --add ARCHIVE FILES...
Create the specified archive with the specified files. You must specify only one archive.
.TP
.B -x, --extract ARCHIVE
Extract all the files on the specified archive. You must specify only one archive.
.TP
.B -l, --list ARCHIVES...
List the content of the specified archives.
.TP
.B -z, --recompress ARCHIVES...
Recompress the specified archives.
If the -1, -2, -3 options are specified is used the smallest file choice from: the
previous compressed data, the new compression and the uncompressed format.
If the -0 option is specified the archive is always rewritten without any compression.
.TP
.B -t, --test ARCHIVES...
Test the specified archives. The tests may be extended with the -p option.
.TP
.B -N, --not-zip
Use the LZMA algorithm when compressing. The generated zips will not be
readable by any other application!
.TP
.B -p, --pedantic
Be pedantic on the zip tests. If this flag is enabled some more extensive test
on the zip integrity are done. These test are generally not done by
other zip utilities.
.TP
.B -0, --shrink-0
Set the compression level to "none". The file are only stored and not compressed.
This option is very useful to expand the archives of .png and .mp3 files. These
files are already strongly compressed, trying to compress them another time is
really a waste of time and resource. This is specifically true for the frontends
that need to access them a lot of times.
.TP
.B -1, --shrink-1
Set the compression level to "normal". This is the default level of compression.
.TP
.B -2, --shrink-2
Set the compression level to "extra". It's SLOW.
.TP
.B -3, --shrink-3
Set the compression level to "extreme". It's VERY SLOW.
.SH COPYRIGHT
Copyright (C) 2002 Andrea Mazzoleni



HISTORY
AdvanceSCAN Version 1.2 2002/08
	) Added the advpng to compress the PNG files.
	) Added the advmng to compress the MNG files.
	) Changed the output format of the advzip -l option.
		Now it's more like unzip -l.
	) Added the noautomake.sh and autogen.sh scripts.

AdvanceSCAN Version 1.1 2002/06
	) Fixed an infinite loop bug testing some small damaged zips.
	) Removed some warning compiling with gcc 3.1.

AdvanceSCAN Version 1.0 2002/05
	) First public release.
	) Fixed the compression percentage computation on big files.
	) The unknow .zip files are now deleted if they contains only
		empty directories.
	) Added the --pedantic option at the advzip utility. These
		tests are only done if requested.
	) Renamed all the "unknow" options in "unknown".
	) Fixed the reading of configuration file containing the
		\r char.
	) Documentation fixes.
	) Other minor fixes.

AdvanceSCAN Version 0.6-beta 2002/05
	) Major revision.
	) Renamed AdvanceSCAN.
	) Updated to the last C++ standard.
	) General cleanup of the code.
	) Added the AdvanceDIFF and AdvanceZIP utility.

MAMESCAN Version 0.5 2001/08
	) Removed the options -fix, -cat, -collision.

MAMESCAN Version 0.4 2000/02
	) Added configure for msdos.

MAMESCAN Version 0.3

MAMESCAN Version 0.2

MAMESCAN Version 0.1 1999/07
	) First version.

MAMESCAN Version 0.0 1998/11
	) Posted in the MAME list the new listinfo command.



doc/advzip.d
Name
	advzip - The AdvanceZIP compressor

Synopsis
	:advzip [-a, --add] [-x, --extract] [-l, --list]
	:[-z, --recompress] [-t, --test] [-0, --shrink-0]
	:[-1, --shrink-1] [-2, --shrink-2] [-3, --shrink-3]
	:[-N, --not- zip] [-p, --pedantic] [-q, --quiet]
	:[-h, --help] [-V, --version] ARCHIVES... [FILES...]

Description
	The main purpose of this utility is to recompress and test
	the zip archives to get the smallest possible size.

	For the recompression the 7-Zip (www.7-zip.com) Deflate
	implementation is used. This implementation generally
	gives 5-10% more compression than the zLib Deflate
	implementation.

	For experimental purpose also the 7-Zip LZMA algorithm is
	available with the -N option. In this case, the generated
	zips WILL NOT BE USEABLE by any other program. To make
	them useable you need to recompress them without the -N
	option. Generally this algorithm gives 10-20% more
	compression than the 7-Zip Deflate implementation.

Options
	-a, --add ARCHIVE FILES...
		Create the specified archive with the specified
		files. You must specify only one archive.

	-x, --extract ARCHIVE
		Extract all the files on the specified archive. You
		must specify only one archive.

	-l, --list ARCHIVES...
		List the content of the specified archives.

	-z, --recompress ARCHIVES...
		Recompress the specified archives. If the -1, -2,
		-3 options are specified is used the smallest file
		choice from: the previous compressed data, the new
		compression and the uncompressed format. If the -0
		option is specified the archive is always rewritten
		without any compression.

	-t, --test ARCHIVES...
		Test the specified archives. The tests may be
		extended with the -p option.

	-N, --not-zip
		Use the LZMA algorithm when compressing. The
		generated zips will not be readable by any other
		application!

	-p, --pedantic
		Be pedantic on the zip tests. If this flag is
		enabled some more extensive test on the zip
		integrity are done. These test are generally not
		done by other zip utilities.

	-0, --shrink-0
		Set the compression level to "none". The file are
		only stored and not compressed. This option is
		very useful to expand the archives of .png and .mp3
		files. These files are already strongly compressed,
		trying to compress them another time is really a
		waste of time and resource. This is specifically
		true for the frontends that need to access them a
		lot of times.

	-1, --shrink-1
		Set the compression level to "normal". This is the
		default level of compression.

	-2, --shrink-2
		Set the compression level to "extra". It's SLOW.

	-3, --shrink-3
		Set the compression level to "extreme". It's VERY
		SLOW.

Copyright
	This file is Copyright (C) 2002 Andrea Mazzoleni

See Also
	advpng(1), advmng(1)
